sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,537 @@
+            elif strategy == "largest cluster":
+                # Pick index of largest cluster by amount of assigned points
+                biggest, _ = max(centers_dict.items(),
+                                 key=lambda x: x[1]['label_indexes'].shape[0])
+            else:
+                # "child_biggest_sse"
+
+                # Pick index of cluster with biggest SSE from latest received
+                # cluster with lower SSE is added first so last cluster
+                # in centers_dict will have higher SSE and be further split
+                biggest = last_center_id
+
+            # Pick data to bisect from selected biggest cluster
+            data_left = X[centers_dict[biggest]['label_indexes']]
+            weights_left = sample_weight[
+                centers_dict[biggest]['label_indexes']]
+
+            # Perform Bisection
+            centers, labels = self._bisect(data_left, init,
+                                           weights_left, random_state)
+
+            if strategy in ["biggest_sse", "child_biggest_sse"]:
+                # Check SSE (Sum of Squared Errors) of each computed centroids.
+                # SSE is calculated with distances between data points
+                # and assigned centroids
+                errors = self._compute_bisect_errors(
+                    data_left, centers, labels, weights_left)
+
+                lower_index = 0 if errors[0] < errors[1] else 1
+                higher_index = 1 if lower_index == 0 else 0
+            else:
+                # "largest_cluster"
+                # SSE of each centroid is not needed here
+                # and won't be calculated
+                errors = None, None
+
+                # Order of labels doesn't matter here
+                lower_index, higher_index = 0, 1
+
+            lower_labels = (labels == lower_index)
+            higher_labels = (labels == higher_index)
+
+            # Add both centroids to dict
+            centers_dict[last_center_id + 1] = {
+                'sse': errors[lower_index],
+                'centroid': centers[lower_index],
+                'label_indexes':
+                    centers_dict[biggest]['label_indexes'][lower_labels]
+            }
+
+            centers_dict[last_center_id + 2] = {
+                'sse': errors[higher_index],
+                'centroid': centers[higher_index],
+                'label_indexes':
+                    centers_dict[biggest]['label_indexes'][higher_labels]
+            }
+
+            if self.verbose:
+                print(f"Centroid Found: {centers[lower_index]}")
+                print(f"Centroid Found: {centers[higher_index]}")
+
+            last_center_id += 2
+
+            # Delete split cluster from dict
+            del centers_dict[biggest]
+
+        # Extract calculated centroids to array
+        # Also save them in self.cluster_centers and self.labels_
+        self._save_labels_and_centers(X.shape[0], centers_dict)
```

Comment:
I thought it should be sligtly faster in that way, due to avoiding unnecessary passing and assigments. 
If you think it will be more valuable that way - I may change that

Reply:
That will be deprecated with next commit